[
  {
    "cmd": [
      "python",
      "-u",
      "[BUILD]/scripts/slave/bot_update.py",
      "--master",
      "tryserver.chromium",
      "--builder",
      "linux_rel",
      "--slave",
      "totallyaslave-c4",
      "--spec",
      "cache_dir = None\nsolutions = [{'deps_file': 'DEPS', 'managed': True, 'name': 'src', 'url': 'svn://svn.chromium.org/chrome/trunk/src'}]",
      "--revision_mapping_file",
      "{}",
      "--patch_url",
      "http://src.chromium.org/foo/bar",
      "--output_json",
      "/path/to/tmp/json",
      "--revision",
      "src@HEAD"
    ],
    "name": "bot_update",
    "~followup_annotations": [
      "step returned non-zero exit code: 1",
      "@@@STEP_TEXT@Some step text@@@",
      "@@@STEP_LOG_LINE@json.output@{@@@",
      "@@@STEP_LOG_LINE@json.output@  \"did_run\": true, @@@",
      "@@@STEP_LOG_LINE@json.output@  \"log_lines\": [@@@",
      "@@@STEP_LOG_LINE@json.output@    [@@@",
      "@@@STEP_LOG_LINE@json.output@      \"patch error\", @@@",
      "@@@STEP_LOG_LINE@json.output@      \"Patch failed to apply\"@@@",
      "@@@STEP_LOG_LINE@json.output@    ]@@@",
      "@@@STEP_LOG_LINE@json.output@  ], @@@",
      "@@@STEP_LOG_LINE@json.output@  \"patch_failure\": true, @@@",
      "@@@STEP_LOG_LINE@json.output@  \"patch_root\": \"src\", @@@",
      "@@@STEP_LOG_LINE@json.output@  \"properties\": {}, @@@",
      "@@@STEP_LOG_LINE@json.output@  \"root\": \"src\", @@@",
      "@@@STEP_LOG_LINE@json.output@  \"step_text\": \"Some step text\"@@@",
      "@@@STEP_LOG_LINE@json.output@}@@@",
      "@@@STEP_LOG_END@json.output@@@",
      "@@@STEP_LOG_LINE@patch error@Patch failed to apply@@@",
      "@@@STEP_LOG_END@patch error@@@"
    ]
  },
  {
    "cmd": [
      "python",
      "-u",
      "import sys\nprint 'Check the bot_update step for details.'\nsys.exit(1)\n"
    ],
    "name": "Patch failure",
    "~followup_annotations": [
      "step returned non-zero exit code: 1",
      "@@@STEP_FAILURE@@@"
    ]
  },
  {
    "name": "$final_result",
    "reason": "Step(Patch failure) failed and was marked as abort_on_failure",
    "status_code": 1
  }
]